pkg/relabel/relabel_test.go

```diff
@@ -84,3 +88,30 @@ func TestRelabel_Run(t *testing.T) {
 		})
 	}
 }
+
+func TestRlabel_NewFromViper(t *testing.T) {
+	t.Run("returns error when yaml config contains unknown keys", func(t *testing.T) {
+		config := []byte(`
+eventRelabelConfigs:
+  eventRelabelConfigs:
+  - source_labels: ["url"]
+    regexP: ".*operationName=(.*)(&.*)?$"
+    target_label: operation_name
+    replacement: "$1" `)
+		viper.SetConfigType("yaml")
+		err := viper.ReadConfig(bytes.NewBuffer(config))
+		if err != nil {
+			t.Fatalf("Unexpected error occured: %s", err)
+		}
+		vc := viper.Sub("EventRelabelConfigs")
+		_, err = NewFromViper(vc, logrus.New())
+		assertError(t, err)
+	})
+}
+
+func assertError(t testing.TB, err error) {
```

Comment:
Same as above, the helper should not be needed https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#Error